src/content/1.7/modules/testing/advanced-checks.md

```diff
@@ -0,0 +1,163 @@
+
+* **Project requires PHP 7 min**
+
+If the current requirements of your project match PHPStan ones, your may add it directly in your project dependencies.
+
+```bash
+composer require --dev phpstan/phpstan
+```
+
+* **Project is compliant with PHP 5.x**
+
+Requiring the library in a project that can run on PHP 5.x will trigger compatibility errors from Composer. One workaround is to install it globally, so it can be run from anywhere flawlessly.
+
+```bash
+composer global require phpstan/phpstan
+```
+
+Another option is to use the Docker image of PHPStan, following the same versioning of the library on Packagist.
+
+```bash
+docker run --rm [...] phpstan/phpstan analyse 
+```
+
+### Setting up
+
+PHPStan can analyse a standalone project without much configuration, where it only needs the files to analyse.
+Some features, like autoloading with composer, are enabled by default.
+
+A module deals with many methods from the core in order to work. To avoid issues like undefined methods and classes, the core needs to be autoloaded in memory each time the module is analysed. PHPStan handles several levels of checks, which are defined in [their documentation page "Rule levels"](https://phpstan.org/user-guide/rule-levels). The higher the better, but somes inconsistencies in the core PHPDoc may prevent you to reach the highest level.
+
+The configuration file stores the parameters that would be sent in the command line otherwise. Having a configuration file is a mandatory step for a module, as we have complex rules to write:
+
+* Files to analyse
+* Level of check to run
+* PrestaShop constants to set as dynamic
+* Potential errors to ignore
+* If run with different versions of PS: Do not report ignored messages that aren't reported.
+
+[Their documentation page "Config reference"](https://phpstan.org/config-reference) explains how to write this configuration file from scratch. We also provide a model having some specificities useful for a module. To setup a project with it, require the package `prestashop/php-dev-tools` and call the initialization of PHPStan:
+
+```bash
+# Install dependencies
+composer require --dev prestashop/php-dev-tools
+
+# Set up configuration files
+php vendor/bin/prestashop-coding-standards phpstan:init
+```
+
+Files will be copied in the folder `tests/phpstan/` of your module. We provide this library to setup an proper environment for a module, for instance to handle constants that are only defined when PrestaShop is installed. If only the source is downloaded in your test environment, the configuration we provide will use an extension file defining these constants for you. In the next version of this library, we expect more rules to be added to avoid false positive issues and add pertinent erros in a module context.
```

Comment:
```suggestion
Files will be copied in the folder `tests/phpstan/` of your module. We provide this library to setup a proper environment for a module, for instance to handle constants that are only defined when PrestaShop is installed. If only the source is downloaded in your test environment, the configuration we provide will use an extension file defining these constants for you. In the next version of this library, we expect more rules to be added to avoid false positive issues and add pertinent errors in a module context.
```